### ISSUE TITLE
Status of Groovy 3 support?

### ISSUE THREAD

**Issue by dmurat** [CodeNarc/CodeNarc#492]
I'm having trouble with several rules, and after some debugging, I believe it is because of differences in AST parsing between Groovy 2 and 3. So I'm wondering what are plans about full Groovy 3 support?

I apologize for asking questions here, but I didn't found pointers to the better place.

Tnx.

**Comment by chrismair** [comment]
We made a bunch of fixes in CodeNarc 1.5 to accommodate Groovy 3.

Could you supply details? Which Groovy version are you running? And provide the violations and if possible, the excerpt of code that caused each? I assume you are using CodeNarc 1.5. Thanks.

**Comment by dmurat** [comment]
Yes, I'm using CodeNarc 1.5, but I also tried some rules on the latest master. I didn't know the status of Groovy 3 support, so I post a general question like this. I can, if appropriate, supply more details in individual issues. At the moment, my main issue is with the Indentation rule since I don't want to turn it off, and I ended up with a bunch of `@SuppressWarnings("Indentation")`.

One thing with the Indentation rule is a warning for the nested class with any kind of modifier (`static`, `private`, etc.). In CodeNarc test suite this causes a failure of several tests in `IndentationRuleTest`, for example `IndentationRuleTest.test_NestedClass_ProperIndentation_NoViolations()`.

Another thing is a warning for method chaining which can be seen with failing `IndentationRuleTest.test_MethodChaining()` test.

I run the tests by modifying test dependencies in `build.gradle`:

```
def groovyVersion = '2.4.17'
def groovyTestVersion = '3.0.3'
...

dependencies {
    ...
    testCompile 'com.github.stefanbirkner:system-rules:1.16.1'

    testRuntimeOnly("org.codehaus.groovy:groovy:$groovyTestVersion")
    testRuntimeOnly("org.codehaus.groovy:groovy-ant:$groovyTestVersion")
    testRuntimeOnly("org.codehaus.groovy:groovy-json:$groovyTestVersion")
    testRuntimeOnly("org.codehaus.groovy:groovy-templates:$groovyTestVersion")
}

```
